### ISSUE TITLE
[Q] how to add files to a finished run?

### ISSUE THREAD

**Issue by Guptajakala** [wandb/wandb#4893]
Hi, I have a run that has finished. But I later have some files to add to it. Is there any way to do it?
I've tried 
```
run.save(file_path)   # run is got from wandb.Api()
```

But this gives error `TypeError: save() takes 1 positional argument but 2 were given`

**Comment by luisbergua** [comment]
Hi @Guptajakala, thanks for writing in! This is because the [run created when calling `wandb.init()` ](https://docs.wandb.ai/ref/python/run) is different that [the one called with wandb.Api().run()](https://docs.wandb.ai/ref/python/public-api/run) so in this case [method `.save()`](https://github.com/wandb/wandb/blob/e85ba97972308091d22f34e275cd58fea4b774dc/wandb/apis/public.py#L1931-L1932) doesn't takes any argument as it's used to save the changes that you may have done (for example updating the config, it basically calls `.update()`) different from [method `.save()`](https://docs.wandb.ai/ref/python/run#save) in `wandb.run()` used to save files. Said this, you cannot save new files to an already finished run as the purpose of the run is to track one concrete experiment so if you have new changes, you need to start a new run to track these changes. It's the same case with metrics, once they're logged, you cannot change their values as this would mean actually modifying the experiment. Please let me know if this explanation makes sense. I can also create a new feature request to be able to add files after the run is finished, just let me know (and if this is the case it'd be great if you could explain me a little bit about your use-case).

**Comment by Guptajakala** [comment]
@luisbergua thanks for the explanation!
One use case for me is after finish training, I want to evaluate on the test set and update those metrics into the same run (For some reason, it's not easy to do training and evaluation in the same script). Is there any existing method to do this? Otherwise adding a feature request would be great!

**Comment by luisbergua** [comment]
Hi @Guptajakala, sorry I was reading the thread again and I think I misunderstood you. You can [resume your run](https://docs.wandb.ai/guides/track/advanced/resuming) and keep logging data to this same run. What is not possible is modify data already logged but after reading your message again I think this is not your case right? Again, sorry for the confusion.

**Comment by Guptajakala** [comment]
@luisbergua Thanks for pointing this. I think it satisfies half of the scenarios. There are still some cases, where I need to modify the logged metrics for old runs (e.g. eval protocol changesd after some weeks, test set changed). I want to have that one place to compare different runs and up-to-date. So this feature is not supported now as I understand?

**Comment by luisbergua** [comment]
@Guptajakala, that's correct, once a metric is logged, this metric cannot be modified. Great, I can create a feature request for this, thanks for your feedback!
